### PR TITLE
[ISSUE #3805] Add `ShenyuService` and `ShenyuDubboService` annotations for shenyu-client-apache-dubbo

### LINKED ISSUE #3805: [WeOpen-Star] Add simpler client annotations with @AliasFor
## How to Participation

1. reply: [WeOpen Star]I would like to help 
2. add this wechat.  note : 摘星
image

### Description

As a downstream of https://github.com/apache/shenyu/issues/3485, this task needs you to add(`do not delete any existing annotations`) more simpler client annotations.

For example, below is how we use `client annotation` now:
```java
@RestController
@RequestMapping("/order")
@ShenyuSpringCloudClient("/order")
public class OrderController {
    
    @PostMapping("/save")
    @ShenyuSpringCloudClient("/save")
    public OrderDTO save(@RequestBody final OrderDTO orderDTO) {
        orderDTO.setName("hello world spring cloud save order");
        return orderDTO;
    }
}
```
Does it seem a little repetitive? We can realize combined annotation through `@AliasFor` and `org.springframework.core.annotation.AnnotatedElementUtils#findMergedAnnotation`.

Notice: the newly added annotation needs to be compatible with all functions of other frameworks !

Use simplified annotations like below:
```java
@RestController
@ShenyuRequestMapping("/order")
public class OrderController {
    
    @ShenyuPostMapping("/save")
    public OrderDTO save(@RequestBody final OrderDTO orderDTO) {
        orderDTO.setName("hello world spring cloud save order");
        return orderDTO;
    }
}
```

You are welcome to make suggestions !

### Task List

1. - [x] add `@ShenyuRequestMapping`, `@ShenyuGetMapping`, `@ShenyuPostMapping`, `@ShenyuPutMapping`, `@ShenyuDeleteMapping`, `@ShenyuPatchMapping` for `shenyu-client-springmvc`; @ShawnJim 
2. - [x] add `@ShenyuRequestMapping`, `@ShenyuGetMapping`, `@ShenyuPostMapping`, `@ShenyuPutMapping`, `@ShenyuDeleteMapping`, `@ShenyuPatchMapping` for `shenyu-client-springcloud`; @ShawnJim 
3. - [x] add `@ShenyuService` for `shenyu-client-alibaba-dubbo`; @mangoGoForward 
4. - [x] add `@ShenyuService`, `@ShenyuDubboService` for `shenyu-client-apache-dubbo`; @mangoGoForward 
5. - [x] add `@ShenyuMotanService` for `shenyu-client-motan`; @runqi-zhao 
6. - [ ] add `@ShenyuTarsServant` for `shenyu-client-tars`; @HeZean 
7. - [x] add `@ShenyuServerEndpoint` for `shenyu-client-websocket`; @lahmXu 

### How to contributor

You can see this : [Contributor Guide](https://shenyu.apache.org/zh/community/contributor-guide/)

1. I suggest you subscribe to shenyu dev mailing list, and when you encounter any problems, or when you finish,
you can send an email to dev@shenyu.apache.org

2.  I suggest you git clone the source code and finish the changes. like this : 

fork  this to you repository :   https://github.com/apache/shenyu.git

```
git clone git@github.com:${YOUR_USERNAME}/${TARGET_REPO}.git

git checkout -b a-dev-branch

git push origin a-dev-branch
```

### Verification

After you modify file in verify local and you can see this : [How to build shenyu](https://github.com/apache/shenyu/blob/master/README.md) .

After you pull request merged. You can verify to this :  [en-doc](https://shenyu.apache.org/docs/next/user-guide/property-config/gateway-property-config) ,  [zh-doc](https://shenyu.apache.org/zh/docs/next/user-guide/property-config/gateway-property-config)

### PR DESCRIPTION
**Pull request by mangoGoForward**
Signed-off-by: mango <xu.weiKyrie@foxmail.com>

Fixes #3805

Make sure that:

- [x] You have read the [contribution guidelines](https://shenyu.apache.org/community/contributor-guide).
- [x] You submit test cases (unit or integration tests) that back your changes.
- [x] Your local test passed `./mvnw clean install -Dmaven.javadoc.skip=true`.

### CODE REVIEW

File: shenyu-client/shenyu-client-dubbo/shenyu-client-apache-dubbo/src/main/java/org/apache/shenyu/client/apache/dubbo/annotation/ShenyuDubboService.java [hunk 1]
```diff
@@ -0,0 +1,446 @@
...
+
+import static org.apache.dubbo.common.constants.CommonConstants.DEFAULT_CALLBACK_INSTANCES;
+import static org.apache.dubbo.common.constants.CommonConstants.DEFAULT_RETRIES;
+
+
+/**
+ * The interface shenyu service.
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ElementType.TYPE, ElementType.METHOD})
```
Comment (dragon-zhang) [inline shenyu-client/shenyu-client-dubbo/shenyu-client-apache-dubbo/src/main/java/org/apache/shenyu/client/apache/dubbo/annotation/ShenyuDubboService.java:41]: remove `ElementType.METHOD`

File: shenyu-client/shenyu-client-dubbo/shenyu-client-apache-dubbo/src/main/java/org/apache/shenyu/client/apache/dubbo/annotation/ShenyuService.java [hunk 2]
```diff
@@ -0,0 +1,446 @@
...
+
+import static org.apache.dubbo.common.constants.CommonConstants.DEFAULT_CALLBACK_INSTANCES;
+import static org.apache.dubbo.common.constants.CommonConstants.DEFAULT_RETRIES;
+
+
+/**
+ * The interface shenyu service.
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ElementType.TYPE, ElementType.METHOD})
```
Comment (dragon-zhang) [inline shenyu-client/shenyu-client-dubbo/shenyu-client-apache-dubbo/src/main/java/org/apache/shenyu/client/apache/dubbo/annotation/ShenyuService.java:41]: remove `ElementType.METHOD`

File: shenyu-examples/shenyu-examples-dubbo/shenyu-examples-apache-dubbo-service-annotation/src/main/java/org/apache/shenyu/examples/apache/dubbo/service/annotation/impl/DubboMultiParamServiceImpl.java [hunk 3]
```diff
@@ -32,7 +32,7 @@
 /**
  * The type Dubbo multi param service.
  */
-@DubboService
+@ShenyuDubboService("demo")
```
Comment (dragon-zhang) [inline shenyu-examples/shenyu-examples-dubbo/shenyu-examples-apache-dubbo-service-annotation/src/main/java/org/apache/shenyu/examples/apache/dubbo/service/annotation/impl/DubboMultiParamServiceImpl.java:35]: rollback this

File: shenyu-examples/shenyu-examples-dubbo/shenyu-examples-apache-dubbo-service-annotation/src/main/java/org/apache/shenyu/examples/apache/dubbo/service/annotation/impl/DubboTestServiceImpl.java [hunk 4]
```diff
@@ -34,7 +34,7 @@
 /**
  * The type Dubbo service.
  */
-@DubboService
+@ShenyuDubboService("demo")
 public class DubboTestServiceImpl implements DubboTestService {
```
Comment (dragon-zhang) [inline shenyu-examples/shenyu-examples-dubbo/shenyu-examples-apache-dubbo-service-annotation/src/main/java/org/apache/shenyu/examples/apache/dubbo/service/annotation/impl/DubboTestServiceImpl.java:38]: rollback this

File: shenyu-examples/shenyu-examples-dubbo/shenyu-examples-spring-cloud-alibaba-dubbo-service-annotation/src/main/java/org/apache/shenyu/examples/spring/cloud/dubbo/service/annotation/impl/MultiParamServiceImpl.java [hunk 5]
```diff
@@ -32,7 +32,7 @@
 /**
  * The type Dubbo multi param service.
  */
-@DubboService
+@ShenyuDubboService("demo")
```
Comment (dragon-zhang) [inline shenyu-examples/shenyu-examples-dubbo/shenyu-examples-spring-cloud-alibaba-dubbo-service-annotation/src/main/java/org/apache/shenyu/examples/spring/cloud/dubbo/service/annotation/impl/MultiParamServiceImpl.java:35]: rollback this

File: shenyu-examples/shenyu-examples-dubbo/shenyu-examples-spring-cloud-alibaba-dubbo-service-annotation/src/main/java/org/apache/shenyu/examples/spring/cloud/dubbo/service/annotation/impl/TestServiceImpl.java [hunk 6]
```diff
@@ -30,7 +30,7 @@
 /**
  * The type Dubbo service.
  */
-@DubboService
+@ShenyuDubboService("test")
```
Comment (dragon-zhang) [inline shenyu-examples/shenyu-examples-dubbo/shenyu-examples-spring-cloud-alibaba-dubbo-service-annotation/src/main/java/org/apache/shenyu/examples/spring/cloud/dubbo/service/annotation/impl/TestServiceImpl.java:33]: rollback this

**Comment by dragon-zhang** [comment]
please also remove `ElementType.METHOD` in `shenyu-client/shenyu-client-dubbo/shenyu-client-alibaba-dubbo/src/main/java/org/apache/shenyu/client/alibaba/dubbo/annotation/ShenyuService.java`, because `dubbo` itself does not support annotated in `method`.

File: shenyu-client/shenyu-client-dubbo/shenyu-client-apache-dubbo/src/main/java/org/apache/shenyu/client/apache/dubbo/annotation/ShenyuDubboService.java (hunk 1 above)
Comment (mangoGoForward) [reply shenyu-client/shenyu-client-dubbo/shenyu-client-apache-dubbo/src/main/java/org/apache/shenyu/client/apache/dubbo/annotation/ShenyuDubboService.java]: Done.

File: shenyu-client/shenyu-client-dubbo/shenyu-client-apache-dubbo/src/main/java/org/apache/shenyu/client/apache/dubbo/annotation/ShenyuService.java (hunk 2 above)
Comment (mangoGoForward) [reply shenyu-client/shenyu-client-dubbo/shenyu-client-apache-dubbo/src/main/java/org/apache/shenyu/client/apache/dubbo/annotation/ShenyuService.java]: Done.

**Comment by mangoGoForward** [comment]
> please also remove `ElementType.METHOD` in `shenyu-client/shenyu-client-dubbo/shenyu-client-alibaba-dubbo/src/main/java/org/apache/shenyu/client/alibaba/dubbo/annotation/ShenyuService.java`, because `@Service` itself does not support annotated in `method`.

Thank for your patient review, already removed.

**Comment by mangoGoForward** [comment]
> please also remove `ElementType.METHOD` in `shenyu-client/shenyu-client-dubbo/shenyu-client-alibaba-dubbo/src/main/java/org/apache/shenyu/client/alibaba/dubbo/annotation/ShenyuService.java`, because `@Service` itself does not support annotated in `method`.

Thank you for your patient review, already removed.

**Comment by mangoGoForward** [comment]
The CI of `build (shenyu-integrated-test-spring-cloud)` failed because can not pull `openjdk:8-jre-alpine` image from docker.io

**Comment by dragon-zhang** [comment]
> The CI of `build (shenyu-integrated-test-spring-cloud)` failed because can not pull `openjdk:8-jre-alpine` image from docker.io

ok, I will rerun failed jobs

**Review by dragon-zhang — APPROVED** [review]

File: shenyu-examples/shenyu-examples-dubbo/shenyu-examples-apache-dubbo-service-annotation/src/main/java/org/apache/shenyu/examples/apache/dubbo/service/annotation/impl/DubboMultiParamServiceImpl.java (hunk 3 above)
Comment (leopardrota) [reply shenyu-examples/shenyu-examples-dubbo/shenyu-examples-apache-dubbo-service-annotation/src/main/java/org/apache/shenyu/examples/apache/dubbo/service/annotation/impl/DubboMultiParamServiceImpl.java]: Why did not this be rollbacked?  It seems that Dubbo only resolves the direct annotation, not composite annotation. (#4031)

File: shenyu-examples/shenyu-examples-dubbo/shenyu-examples-apache-dubbo-service-annotation/src/main/java/org/apache/shenyu/examples/apache/dubbo/service/annotation/impl/DubboTestServiceImpl.java (hunk 4 above)
Comment (leopardrota) [reply shenyu-examples/shenyu-examples-dubbo/shenyu-examples-apache-dubbo-service-annotation/src/main/java/org/apache/shenyu/examples/apache/dubbo/service/annotation/impl/DubboTestServiceImpl.java]: Why did not this be rollbacked? It seems that Dubbo only resolves the direct annotation, not composite annotation.

File: shenyu-examples/shenyu-examples-dubbo/shenyu-examples-spring-cloud-alibaba-dubbo-service-annotation/src/main/java/org/apache/shenyu/examples/spring/cloud/dubbo/service/annotation/impl/MultiParamServiceImpl.java (hunk 5 above)
Comment (leopardrota) [reply shenyu-examples/shenyu-examples-dubbo/shenyu-examples-spring-cloud-alibaba-dubbo-service-annotation/src/main/java/org/apache/shenyu/examples/spring/cloud/dubbo/service/annotation/impl/MultiParamServiceImpl.java]: Why was this not rollbacked? It seems that Dubbo only resolves the direct annotation, not composite annotation.

File: shenyu-examples/shenyu-examples-dubbo/shenyu-examples-spring-cloud-alibaba-dubbo-service-annotation/src/main/java/org/apache/shenyu/examples/spring/cloud/dubbo/service/annotation/impl/TestServiceImpl.java (hunk 6 above)
Comment (leopardrota) [reply shenyu-examples/shenyu-examples-dubbo/shenyu-examples-spring-cloud-alibaba-dubbo-service-annotation/src/main/java/org/apache/shenyu/examples/spring/cloud/dubbo/service/annotation/impl/TestServiceImpl.java]: Why was this not rollbacked? It seems that Dubbo only resolves the direct annotation, not composite annotation.

File: shenyu-examples/shenyu-examples-dubbo/shenyu-examples-apache-dubbo-service-annotation/src/main/java/org/apache/shenyu/examples/apache/dubbo/service/annotation/impl/DubboMultiParamServiceImpl.java (hunk 3 above)
Comment (dragon-zhang) [reply shenyu-examples/shenyu-examples-dubbo/shenyu-examples-apache-dubbo-service-annotation/src/main/java/org/apache/shenyu/examples/apache/dubbo/service/annotation/impl/DubboMultiParamServiceImpl.java]: > Why was this not rollbacked? It seems that Dubbo only resolves the direct annotation, not composite annotation. (#4031)

please check the `master` branch, it has benn rollbacked.

**Pull request merged**